web-app/js/portal/filter/ui/FilterGroupPanel.js

```diff
@@ -147,34 +148,32 @@ Portal.filter.ui.FilterGroupPanel = Ext.extend(Ext.Container, {
         this._addErrorMessage(OpenLayers.i18n('subsetEmptyFiltersText'));
     },
 
-    _sortPanels: function(panels) {
+    _sortFilters: function(filters) {
 
         var panelOrder = [
-            Portal.filter.ui.GeometryFilterService,
-            Portal.filter.ui.DateFilterPanel,
-            Portal.filter.ui.BooleanFilterPanel,
-            Portal.filter.ui.NumberFilterPanel,
-            Portal.filter.ui.ComboFilterPanel
+            Portal.filter.GeometryFilter,
+            Portal.filter.DateFilter,
+            Portal.filter.BooleanFilter,
+            Portal.filter.NumberFilter,
+            Portal.filter.StringFilter
         ];
 
-        var typeOrder = function (panel) {
-            return panelOrder.indexOf(panel.constructor) * -1;
+        var typeOrder = function (filter) {
+            return panelOrder.indexOf(filter.constructor) * -1;
         };
 
         var _this = this;
-        panels.sort(function(firstPanel, secondPanel) {
-            var comparisonResult = _this._compareElements(typeOrder(firstPanel), typeOrder(secondPanel));
+        filters.sort(function(firstFilter, secondFilter) {
```

Comment:
I'd like the logic of this function to become something simpler, perhaps also not in a closure.

It should be something along the lines of:
```
compareFilterOrder: function(lhs, rhs) {
    return compareType(lhs, rhs) || compareLabel(lhs, rhs);
},

compareType: function(lhs, rhs) {
    if (lhs.type == rhs.type) {
        return false;
    }
    else {
        return lhs > rhs ? lhs : rhs;
    }
},

compareLabel: function(lhs, rhs) {
    return lhs > rhs ? lhs : rhs;
}
```

I'm not a big fan of the `-1` multiplier. It's very vague.